neurodamus/gj_user_corrections.py

```diff
@@ -164,68 +164,86 @@ def _update_gpas(node_manager, filename, gjc, correction_iteration_load):
     try:
         holding_per_gid = h5py.File(filename, "r")
     except OSError:
         raise ConfigurationError(f"Error opening MEComboInfo file {filename}")
+    if f"holding_per_gid/{gjc}" not in holding_per_gid:
+        logging.warning(f"Data for holding_per_gid/{gjc} not found in {holding_per_gid}")
+        return holding_ic_per_gid
     raw_cell_gids = node_manager.local_nodes.raw_gids()
     offset = node_manager.local_nodes.offset
     for agid in holding_per_gid["holding_per_gid"][str(gjc)]:
         gid = int(agid[1:])
         if gid in raw_cell_gids:
-            holding_ic_per_gid[gid] = Nd.h.IClamp(
-                0.5, sec=node_manager.getCell(gid + offset).soma[0]
+            final_gid = gid + offset
+            holding_ic_per_gid[final_gid] = Nd.h.IClamp(
+                0.5, sec=node_manager.getCell(final_gid).soma[0]
             )
-            holding_ic_per_gid[gid].dur = 9e9
-            holding_ic_per_gid[gid].amp = holding_per_gid["holding_per_gid"][str(gjc)][agid][()]
+            holding_ic_per_gid[final_gid].dur = 9e9
+            try:
+                holding_ic_per_gid[final_gid].amp = holding_per_gid["holding_per_gid"][str(gjc)][agid][()]  # noqa: E501 #fmt: skip
+            except Exception as e:
+                raise ConfigurationError(
+                    f"Failed to load data in g_pas file {filename}: {e}"
+                ) from e
     return holding_ic_per_gid
 
 
 def _find_holding_current(node_manager, filename):
+    """Add SEClamps to certain cells for holding the membrane voltage
+    Cell ids, voltage amplitude are provided in the input file
+    """
     import h5py
 
     try:
         v_per_gid = h5py.File(filename, "r")
     except OSError:
         raise ConfigurationError(f"Error opening voltage file {filename}")
+
+    logging.info("Inject V_Clamp without disabling holding current!")
+
     seclamp_per_gid = {}
-    seclamp_current_per_gid = {}
     raw_cell_gids = node_manager.local_nodes.raw_gids()
     offset = node_manager.local_nodes.offset
     for agid in v_per_gid["v_per_gid"]:
         gid = int(agid[1:])
         if gid in raw_cell_gids:
-            seclamp_per_gid[gid] = Nd.h.SEClamp(0.5, sec=node_manager.getCell(gid + offset).soma[0])
-            seclamp_per_gid[gid].dur1 = 9e9
-            seclamp_per_gid[gid].amp1 = float(v_per_gid["v_per_gid"][agid][()])
-            seclamp_per_gid[gid].rs = 0.0000001
-            seclamp_current_per_gid[gid] = Nd.h.Vector()
-            seclamp_current_per_gid[gid].record(seclamp_per_gid[gid]._ref_i)
-    v_per_gid.close()
-    return seclamp_current_per_gid
-
-
-def _save_seclamps(seclamp_current_per_gid, output_dir):
-    logging.info("Saving SEClamp Data")
-    seclamp_current_per_gid_a = {}
-    for gid in seclamp_current_per_gid:
-        seclamp_current_per_gid_a[gid] = np.array(seclamp_current_per_gid[gid])
-    pickle.dump(seclamp_current_per_gid_a, open(f"{output_dir}/data_for_host_{MPI.rank}.p", "wb"))
```

Comment:
Hello Oren @orena1, how are you?
I am thinking to remove the `save_seclamps` step in your gap junction correction script. As I understand, it tries to dump the seclamps' currents to a file, but it dumps before we launch the neuron simulation.  As I check, the numpy array is empty.
What did you intend to do for this step? Should it be called after the neuron solver? 
If you think it is not needed (FYI, it was not used in Elisabetta's thalamus simulation), I will take it out.

Reply:
Hi, I think it is needed, adding gap junction change the seady state volate of the cells. So we need to recalcualte the current that need to be injected to each cell in additional to the one that is already injected. (There is some value that is injected per cell I think it is the results of the single neuron optimization)